src/mixins/button-group.js

```diff
@@ -45,36 +53,72 @@ export default {
       if (selected.length === 1) {
         this.buttons[selected[0]].$el.setAttribute('data-only-child', true)
       }
+
+      this.ensureMandatoryInvariant(selected.length > 0)
     },
     register (button) {
       const index = this.buttons.length
       this.buttons.push(button)
       this.listeners.push(this.updateValue.bind(this, index))
       button.$on('click', this.listeners[index])
     },
-    unregister (button) {
-      const index = this.buttons.indexOf(button)
-      if (index === -1) {
+    unregister (buttonToUnregister) {
+      // Basic cleanup if we're destroying
+      if (this.destroying) {
+        const index = this.buttons.indexOf(buttonToUnregister)
+        if (index !== -1) {
+          buttonToUnregister.$off('click', this.listeners[index])
+        }
         return
       }
 
-      const wasSelected = this.isSelected(index)
+      this.redoRegistrations(buttonToUnregister)
+    },
+    redoRegistrations (buttonToUnregister) {
+      let selectedCount = 0
+
+      const buttons = []
+      for (let index = 0; index < this.buttons.length; ++index) {
+        const button = this.buttons[index]
+        if (button !== buttonToUnregister) {
+          buttons.push(button)
+          if (this.isSelected(index)) {
+            ++selectedCount
+          }
+        }
 
-      button.$off('click', this.listeners[index])
-      this.buttons.splice(index, 1)
-      this.listeners.splice(index, 1)
+        button.$off('click', this.listeners[index])
+      }
 
+      this.buttons = []
+      this.listeners = []
+
+      for (let index = 0; index < buttons.length; ++index) {
+        const button = buttons[index]
```

Comment:
does this need to be declared or can we simply pass `buttons[index]` to `this.register`?